File: app/src/main/java/com/example/myapplication/data/Book.kt
package com.example.myapplication.data

data class Book(
    val id: String,
    val title: String,
    val author: String,
    val type: String,
    val year: Int,
    val language: String,
    val theme: String,
    val edition: String? = null,
    val synopsis: String? = null,
    val coverUrl: String? = null,
    val availableCopies: Int = 0,
    val sector: String? = null,
    val shelfCode: String? = null,
    )

File: biblioteca-backend/src/models/Book.js
import mongoose from 'mongoose';

const BookSchema = new mongoose.Schema({
  title: { type: String, required: true },
  author: { type: String, required: true },
  isbn: { type: String },
  copiesTotal: { type: Number, default: 1 },
  copiesAvailable: { type: Number, default: 1 },
  tags: [{ type: String }],
  coverUrl: { type: String },
  description: { type: String },
  sector: { type: String },
  shelfCode: { type: String },
  year: { type: Number },
  edition: { type: String }
}, { timestamps: true });

export default mongoose.model('Book', BookSchema);